File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let prefix = "";
    if(strs.length===0)return prefix;

    for(let i=0;i<strs[0].length;i++){
        //check character by character of first word
        let char = strs[0][i];
        for(let j=0;j<strs.length;j++){
            //if at any point it doesn't match return prefix
            if(strs[j][i]!=char)return prefix;
        }
        prefix+=char;
    }
    return prefix;
};  

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = ""
        if not strs:  # Equivalent to `if(strs.length === 0) return prefix;`
            return prefix

        for i in range(len(strs[0])):  # Loop through characters of the first word
            char = strs[0][i]
            for j in range(len(strs)):  # Loop through all words
                if i >= len(strs[j]) or strs[j][i] != char:  # Ensure index is in bounds
                    return prefix
            prefix += char  # Append matching character
    
        return prefix